### ISSUE TITLE
ItemsAdder packets not being sent; Resource Packs will not download correctly

### ISSUE THREAD

**Issue by Mike5357** [IzzelAliz/Arclight#596]
### I have confirmed that ...

- [X] Arclight is up to date
- [X] all dependencies are installed
- [X] all plugins and mods are up to date
- [X] unable to reproduce in Spigot
- [X] unable to reproduce in Forge

### Arclight version

arclight-1.16.5-1.0.23-SNAPSHOT-2dda792d

### OS & Java versions

Java 11.0.12

### Plugins and Mods

```raw
Plugins: ItemsAdder 3.1.6, ProtocolLib 4.7.0, LoneLibs v1.0.18
```

### Description

ResourcePacks fail to be properly sent to the client due to 
```
[ItemsAdder] Unhandled exception occured in onPacketSending(PacketEvent) for ItemsAdder
java.lang.ClassCastException: class net.minecraft.util.text.ChatType cannot be cast to class [Lnet.md_5.bungee.api.chat.BaseComponent; (net.minecraft.util.text.ChatType and [Lnet.md_5.bungee.api.chat.BaseComponent; are in unnamed module of loader cpw.mods.modlauncher.TransformingClassLoader @b61edb9)
	at dev.lone.itemsadder.main.ii.onPacketSending(SourceFile:116) ~[?:?]
	at com.comphenix.protocol.injector.SortedPacketListenerList.invokeSendingListener(SortedPacketListenerList.java:195) ~[?:?]
	at com.comphenix.protocol.injector.SortedPacketListenerList.invokePacketSending(SortedPacketListenerList.java:149) ~[?:?]
	at com.comphenix.protocol.injector.PacketFilterManager.handlePacket(PacketFilterManager.java:535) ~[?:?]
	at com.comphenix.protocol.injector.PacketFilterManager.invokePacketSending(PacketFilterManager.java:516) ~[?:?]
	at com.comphenix.protocol.injector.netty.ProtocolInjector.packetQueued(ProtocolInjector.java:346) ~[?:?]
	at com.comphenix.protocol.injector.netty.ProtocolInjector.onPacketSending(ProtocolInjector.java:308) ~[?:?]
	at com.comphenix.protocol.injector.netty.ChannelInjector.processSending(ChannelInjector.java:433) ~[?:?]
	at com.comphenix.protocol.injector.netty.ChannelInjector.access$800(ChannelInjector.java:70) ~[?:?]
	at com.comphenix.protocol.injector.netty.ChannelInjector$3.handleScheduled(ChannelInjector.java:373) ~[?:?]
	at com.comphenix.protocol.injector.netty.ChannelInjector$3.onMessageScheduled(ChannelInjector.java:343) ~[?:?]
	at com.comphenix.protocol.injector.netty.ChannelProxy$2.schedulingRunnable(ChannelProxy.java:127) ~[?:?]
	at com.comphenix.protocol.injector.netty.EventLoopProxy.execute(EventLoopProxy.java:95) ~[?:?]
	at net.minecraft.network.NetworkManager.func_150732_b(NetworkManagerMixin.java:191) ~[?:?]
	at net.minecraft.network.NetworkManager.func_201058_a(NetworkManagerMixin.java:163) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_211148_a(ServerPlayNetHandlerMixin.java:1052) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_147359_a(ServerPlayNetHandlerMixin.java:1035) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.entity.CraftPlayer.sendRawMessage(CraftPlayerMixin.java:200) ~[arclight:arclight-1.16.5-1.0.23-daa3c8bb]
	at org.bukkit.craftbukkit.v1_16_R3.entity.CraftPlayer.sendMessage(CraftPlayerMixin.java:216) ~[arclight:arclight-1.16.5-1.0.23-daa3c8bb]
	at dev.lone.itemsadder.main.iO.b(SourceFile:31) ~[?:?]
	at dev.lone.itemsadder.main.id.aj(SourceFile:34) ~[?:?]
	at dev.lone.itemsadder.main.aB.a(SourceFile:2358) ~[?:?]
	at dev.lone.itemsadder.main.hH.a(SourceFile:286) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:81) [arclight:arclight-1.16.5-1.0.23-daa3c8bb]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54) [arclight:arclight-1.16.5-1.0.23-daa3c8bb]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

### Step to reproduce

1. Install ItemsAdder, ProtocolLib, and LoneLibs
2. Open ItemsAdder/config.yml and set self-host enabled to true
3. Type /iazip to force the server to send the Resource Pack to your client (Click Accept) - This is where the console spams `class net.minecraft.util.text.ChatType cannot be cast to class [Lnet.md_5.bungee.api.chat.BaseComponent` that can be seen in the attached logs and the resource pack is not properly applied (all assets such as fonts are missing, as can be seen immediately by the missing font icons above the hotbar:)
![image](https://user-images.githubusercontent.com/12630858/170848505-be241c5d-be8f-4970-8e6b-eb3945ef9189.png)

### Logs

[latest.log](https://github.com/IzzelAliz/Arclight/files/8792483/latest.log)

**Comment by thewickening** [comment]
I can confirm, that I have the exact same issues as stated above.

Would be amazing if this could be fixed, as this is one of the best plugins to accomplish what many server owners in the Pixelmon community are wanting to do.

**Comment by IzzelAliz** [comment]
Fixed in 7debdef4a1958377fd0c1d5f61c1a7ade8eb4956

**Comment by Mike5357** [comment]
Can confirm this fix fixed the error, thanks! However the part where ResourcePacks are not downloading correctly is actually being caused by a separate Mod and not an incompatibility with arclight, so this ticket can now be closed :)
